bugsnag-plugin-android-ndk/src/main/jni/jnicache.c

```diff
@@ -0,0 +1,249 @@
+//
+// Created by Karl Stenerud on 22.02.21.
+//
+
+#include "jnicache.h"
+#include <stdatomic.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <bugsnag_ndk.h>
+#include "safejni.h"
+
+static type_bsg_jni_cache jni_cache;
+static bool is_cache_valid = false;
+
+static inline jclass safe_find_class(JNIEnv *env, const char *name) {
```

Comment:
methods should be prefixed with `bsg` throughout the changeset to avoid any potential for symbol clashes

Reply:
It's a static method, so it won't clash.